internal/worker_runner.go

```diff
@@ -537,12 +539,19 @@ func (sq *SignalsQueue) lastEnqueuedTime(key string) (time.Time, bool) {
 	defer sq.Unlock()
 	t, ok := sq.lastTimeChecked[key]
 	return t, ok
-
 }
 
 func (sq *SignalsQueue) Enqueue(signal models.SignalData) {
 	sq.Lock()
 	defer sq.Unlock()
+	// only enqueue limit freq signals once if same value
+	if signal.LimitFrequency {
```

Comment:
hm, it might be a lot runtime complexity. Maybe add some map, which will hold the names of the signals accumulated for the cycle?

Reply:
yes good point, could change sq.signals to be a map[string]etc  and then that lookup would be a lot quicker.